extensions/azurePublishNew/src/recoilModel/atoms/resourceConfigurationState.ts

```diff
@@ -14,6 +14,8 @@ export const resourceConfigurationState = atom<ResourceConfigurationState>({
     tenantId: '',
     deployLocation: '',
     luisRegion: '',
+    isNewResourceGroup: false,
+    hostName: '',
```

Comment:
You don't have to have a single state atom for all the things across the wizard.  Recoil is built for distributed state, so you could have them as individual atoms.  This way you don't get re-renders when one control updates a single value.